src/app/vitessce-grid-utils.js

```diff
@@ -103,20 +103,14 @@ export function createLoaders(datasets, configDescription) {
     };
     dataset.files.forEach((file) => {
       // Fall back to JsonLoader if a loader is not found for the file type.
-      const matchingLoaderClass = fileTypeToLoader[file.fileType] || JsonLoader;
-      let loader;
-      if (Array.isArray(matchingLoaderClass)) {
-        const [DataSourceClass, LoaderClass] = matchingLoaderClass;
-        // Create _one_ BaseLoaderClass instance per URL. Derived loaders share this object.
-        if (!(file.url in dataSources)) {
-          const { url, requestInit } = file;
-          dataSources[file.url] = new DataSourceClass({ url, requestInit });
-        }
-        loader = new LoaderClass(dataSources[file.url], file);
-      } else {
-        // eslint-disable-next-line new-cap
-        loader = new matchingLoaderClass(file);
+      const matchingLoaderClass = fileTypeToLoaderAndSource[file.fileType] || JsonLoader;
```

Comment:
```suggestion
      const matchingLoaderClass = fileTypeToLoaderAndSource[file.fileType] || [JsonSource, JsonLoader];
```

Since you destruct this below into a 2-element array. May also need an import above for `JsonSource`